contracts/modules/Checkpoint/EtherDividendCheckpoint.sol

```diff
@@ -53,53 +25,54 @@ contract EtherDividendCheckpoint is ICheckpoint, Module {
-        uint256 dividendIndex = dividends.length;
+    function createDividendWithCheckpoint(uint256 _maturity, uint256 _expiry, uint256 _checkpointId) payable external onlyOwner {
+        createDividendWithCheckpointAndExclusions(_maturity, _expiry, _checkpointId, excluded);
+    }
+
+    /**
+     * @notice Creates a dividend and checkpoint for the dividend, specifying explicit excluded addresses
+     * @param _maturity Time from which dividend can be paid
+     * @param _expiry Time until dividend can no longer be paid, and can be reclaimed by issuer
+     * @param _excluded List of addresses to exclude
+     */
+    function createDividendWithExclusions(uint256 _maturity, uint256 _expiry, address[] _excluded) payable public onlyOwner {
         uint256 checkpointId = ISecurityToken(securityToken).createCheckpoint();
-        uint256 currentSupply = ISecurityToken(securityToken).totalSupply();
-        dividends.push(
-          Dividend(
-            checkpointId,
-            now,
-            _maturity,
-            _expiry,
-            msg.value,
-            0,
-            currentSupply,
-            false
-          )
-        );
-        emit EtherDividendDeposited(msg.sender, checkpointId, now, _maturity, _expiry, msg.value, currentSupply, dividendIndex);
+        createDividendWithCheckpointAndExclusions(_maturity, _expiry, checkpointId, _excluded);
     }
 
     /**
-     * @notice Creates a dividend with a provided checkpoint
+     * @notice Creates a dividend with a provided checkpoint, specifying explicit excluded addresses
      * @param _maturity Time from which dividend can be paid
      * @param _expiry Time until dividend can no longer be paid, and can be reclaimed by issuer
      * @param _checkpointId Id of the checkpoint from which to issue dividend
+     * @param _excluded List of addresses to exclude
      */
-    function createDividendWithCheckpoint(uint256 _maturity, uint256 _expiry, uint256 _checkpointId) payable public onlyOwner {
-        require(_expiry > _maturity);
-        require(_expiry > now);
-        require(msg.value > 0);
+    function createDividendWithCheckpointAndExclusions(uint256 _maturity, uint256 _expiry, uint256 _checkpointId, address[] _excluded) payable public onlyOwner {
+        require(_excluded.length <= EXCLUDED_ADDRESS_LIMIT, "Too many addresses excluded");
+        require(_expiry > _maturity, "Expiry is before maturity");
+        require(_expiry > now, "Expiry is in the past");
+        require(msg.value > 0, "No dividend sent");
         require(_checkpointId <= ISecurityToken(securityToken).currentCheckpointId());
         uint256 dividendIndex = dividends.length;
         uint256 currentSupply = ISecurityToken(securityToken).totalSupplyAt(_checkpointId);
+        uint256 excludedSupply = 0;
+        for (uint256 i = 0; i < _excluded.length; i++) {
+            excludedSupply = excludedSupply.add(ISecurityToken(securityToken).balanceOf(_excluded[i]));
+        }
```

Comment:
Shouldn't be here `excludedSupply.add(ISecurityToken(securityToken).balanceOfAt(_excluded[i]));` ?

Reply:
Fixed